bravado/http_client.py

```diff
@@ -197,6 +197,9 @@ def __init__(self, session, request):
         self.session = session
         self.request = request
 
+    def result(self, *args, **kwargs):
```

Comment:
Hmm, I thought it used to be, but I don't see it any more, maybe it's been fixed already?

I guess the only issue now is the need for the branch logic, instead of passing along a Future that already has the correct adapter from the `request()` call.